pkg/operator/ceph/cluster/mgr/dashboard.go

```diff
@@ -110,58 +105,96 @@ func (c *Cluster) configureDashboardModules() error {
 	return nil
 }
 
-func (c *Cluster) configureDashboardModuleSettings(daemonID string) (bool, error) {
+// Delete the manager per-daemon configuration. Returns true
+// if all the configuration entries have been delete successfully.
+func (c *Cluster) deleteManagerDaemonConfiguration() bool {
+
+	mgrKeysToDelete := []string{
+		"mgr/dashboard/url_prefix",
+		"mgr/dashboard/ssl",
+		"mgr/dashboard/PROMETHEUS_API_HOST",
+		"mgr/dashboard/PROMETHEUS_API_SSL_VERIFY",
+		"mgr/dashboard/server_port",
+		"mgr/dashboard/ssl_server_port",
+	}
+
+	success := true
 	monStore := config.GetMonStore(c.context, c.clusterInfo)
+	for _, daemonID := range c.getDaemonIDs() {
+		mgrDaemonID := fmt.Sprintf("%s.%s", config.MgrType, daemonID)
+		for _, key := range mgrKeysToDelete {
+			err := monStore.Delete(mgrDaemonID, key)
+			if err != nil {
+				success = false
```

Comment:
Let's add a log statement to show which key failed and what the error was

Reply:
@travisn I had that log but then removed it because it's already logged by the Delete function itself as you can see in the code below. I just improved that function by adding the `who` also because it was only logging the option.

```
func (m *MonStore) Delete(who, option string) error {
	logger.Infof("deleting %q %q option from the mon configuration database", who, option)
	args := []string{"config", "rm", who, normalizeKey(option)}
	cephCmd := client.NewCephCommand(m.context, m.clusterInfo, args)
	out, err := cephCmd.RunWithTimeout(exec.CephCommandsTimeout)
	if err != nil {
		return errors.Wrapf(err, "failed to delete ceph config in the centralized mon configuration database. output: %s",
			string(out))
	}

	logger.Infof("successfully deleted %q option from the mon configuration database", option)
	return nil
}

```

Reply:
Is that error logged? It looks like if err is returned, it just sets `success = false` and continues. We don't know what the failure was.

Reply:
@travisn the error is `logged` by the Delete function itself:

```
		for _, key := range mgrKeysToDelete {
			err := monStore.Delete(mgrDaemonID, key)
```
and at the end we log whether all the deletes went OK or there was some failure by: 

```
	if success {
		logger.Info("All per-daemon mgr configuration has been deleted successfully.")
	} else {
		logger.Error("At least one delete operation failed while trying to delete per-daemon mgr configuration.")
	}
```

I think from the point of view of logging we should be OK.

Reply:
Whenever there is a failure, we need to log the specific `err`, or else we really don't know how to troubleshoot. So if we log the specific err here on this line, we don't need to log the general error on line 136.

Reply:
@travisn this already done on the `Delete` function:

```
	out, err := cephCmd.RunWithTimeout(exec.CephCommandsTimeout)
	if err != nil {
		return errors.Wrapf(err, "failed to delete ceph config in the centralized mon configuration database. output: %s",
			string(out))
	}
```
Not sure if I'm following at this point... but if add another log we will be just duplicating the error message!

Reply:
The err is returned, but I don't see where the `err` var is logged. It appears the `err` result is ignored other than setting `success = false`, right?

Reply:
@travisn you are right, not sure why I thought that the error was being logged inside the Delete when reading the code again and again. It's just being returned. I'll fix it.